ci.sh

```diff
@@ -556,6 +556,8 @@ cmd_coverage_report() {
     --filter '.*jxl/.*'
     --exclude '.*_test.cc'
     --exclude '.*_testonly..*'
+    --exclude '.*_debug.*'
```

Comment:
this ships in the library, not sure it is fair to ignore it.

Reply:
The _debug stuff is behind an `if (constexpr false)`, so I don't think that really counts as shipping.

The aux_out stuff: I suppose some of the bookkeeping code (attribution of encoded bits to bitstream segments) does perhaps ship with the encoder, but most of the code there is for printing verbose output and dumping debug images, and those functions should only be called from cjxl and benchmark_xl, so they shouldn't end up in the library or be accessible in any way from the API.

Reply:
Shouldn't attribution of bits be tested?

Reply:
Bit attribution is only used to show verbose encoder output, showing how many bits went to header, toc, DC, etc. I don't know if we should keep that functionality - and if we do, it's probably best put behind a macro that can be disabled in regular builds. In any case, I don't think this is info we want to expose in the api.

Reply:
ok with not having it in the API; but I don't follow from there that it shouldn't be in the coverage report. What if the bit attribution is buggy?

If this PR is just to improve the % value in the coverage then please don't merge this PR.

Not reporting test infrastructure is one thing, because we don't test the test infrastructure and we don't intend to do that. On the other hand `lib/extras` for example has some tests (not reported here); and bugs parsing a .gif file in cjxl would be reported in this repo as issues when people try cjxl with a .gif, so it is fair to get that coverage even if codec_gif is not something in our API.

If the bit attribution in the encoder is something we want remove, then removing it is the right solution, not hiding it in the coverage report. I think that the bit attribution can be useful information when optimizing the encoder so it should stay (and have some test while at that).

Reply:
Fair enough. I'll change this PR to keep aux_out in the coverage report.

Bit attribution and dumping debug images can certainly be useful when debugging/optimizing the encoder, but it should not be compiled into normal builds of libjxl imo. Also if/when we eventually migrate cjxl/benchmark_xl to the api, we'll likely have to do it in a different way anyway. But you're right that hiding it from the coverage report doesn't solve anything, and it is better to resolve the underlying problem rather than just hiding it.

I'm not sure what the best approach is here, but I think it might be best to put all the bit attribution calls behind a macro (like we do with profiling, which is also something that inherently requires a presence in many source files), and turn it into no-op on normal builds and only make it do something in debug builds.